src/query/graphite/storage/converter.go

```diff
@@ -58,10 +59,25 @@ func glob(metric string) []byte {
 }
 
 func convertMetricPartToMatcher(count int, metric string) models.Matcher {
+	var (
+		matchType models.MatchType
+		value     []byte
+	)
+
+	// NB: small optimization to use exact matcher rather than regex if only
+	// alphanumeric characters present.
```

Comment:
Wouldn't it make more sense to do always try to do a glob and make glob return `([]byte, bool)` and if it returns `false` to imply that it didn't need to turn the glob into a regexp then you can use a MatchEqual matcher?

I'm pretty sure that's how we used to do it (and helps in the case you use a non alphanumeric symbol but turns out that it's still not a regexp? (i.e. using the `-` or `_` character).